examples/nextjs-with-typescript/src/Link.tsx

```diff
@@ -1,22 +1,19 @@
 /* eslint-disable jsx-a11y/anchor-has-content */
-import React from 'react';
+import * as React from 'react';
 import clsx from 'clsx';
 import { useRouter } from 'next/router';
 import NextLink from 'next/link';
-import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import MuiLink from '@material-ui/core/Link';
+import { TypographyProps } from '@material-ui/core/Typography';
 
 interface NextLinkProps {
   as?: string;
   href?: string;
   prefetch?: boolean;
+  className?: string;
 }
 
-type NextComposedProps = React.AnchorHTMLAttributes<HTMLAnchorElement> & NextLinkProps;
-
-const NextComposed = React.forwardRef(function NextComposed(
-  props: NextComposedProps,
-  ref: React.Ref<any>,
-) {
+const NextComposed = React.forwardRef<HTMLAnchorElement, NextLinkProps>((props, ref) => {
```

Comment:
This changes the display name in react devtools:

3.6:
```diff
- ForwardRef(NextComposed)
+ ForwardRef
```

experimental:
```diff
- <NextComposed> ForwardRef
+ <Anonymous> ForwardRef
```

Both of which are less ideal.

Reply:
Can have is this way:

```typescript
const NextComposed = React.forwardRef<HTMLAnchorElement, NextLinkExtendedProps>(
  function NextComposed(props, ref) {
```

That way dev tools will show `ForwardRef(NextComposed)`

Personally I think it is confusing having `NextComposed` being both a constant and a function, but that achieves the goal if dev tools are higher priority.